--- Makefile
+++ Makefile
@@ -73,6 +73,8 @@
 # CertManager is installed by default; skip with:
 # - CERT_MANAGER_INSTALL_SKIP=true
 KIND_CLUSTER ?= podbeacon-test-e2e
+KUBECONFIG ?= $(shell pwd)/bin/kubeconfig-e2e
+export KUBECONFIG
 
 .PHONY: setup-test-e2e
 setup-test-e2e: ## Set up a Kind cluster for e2e tests if it does not exist
@@ -85,15 +87,17 @@
 			echo "Kind cluster '$(KIND_CLUSTER)' already exists. Skipping creation." ;; \
 		*) \
 			echo "Creating Kind cluster '$(KIND_CLUSTER)'..."; \
-			$(KIND) create cluster --name $(KIND_CLUSTER) --image kindest/node:v1.31.0 ;; \
+			$(KIND) create cluster --name $(KIND_CLUSTER) --image kindest/node:v1.33.0 --kubeconfig $(KUBECONFIG); \
+			touch $(shell pwd)/bin/.kind_cluster_created ;; \
 	esac
+	@$(KIND) export kubeconfig --name $(KIND_CLUSTER) --kubeconfig $(KUBECONFIG)
 
 .PHONY: test-e2e
 test-e2e: setup-test-e2e manifests generate fmt vet ## Run the e2e tests. Expected an isolated environment using Kind.
 	KIND=$(KIND) KIND_CLUSTER=$(KIND_CLUSTER) go test -tags=e2e ./test/e2e/ -v -ginkgo.v
-	$(MAKE) cleanup-test-e2e
 
 .PHONY: cleanup-test-e2e
 cleanup-test-e2e: ## Tear down the Kind cluster used for e2e tests
-	@$(KIND) delete cluster --name $(KIND_CLUSTER)
+	@if [ -f $(shell pwd)/bin/.kind_cluster_created ]; then \
+		$(KIND) delete cluster --name $(KIND_CLUSTER) --kubeconfig $(KUBECONFIG); \
+		rm -f $(shell pwd)/bin/.kind_cluster_created $(KUBECONFIG); \
+	fi
